src/states/common/unrelocated.rs

```diff
@@ -0,0 +1,64 @@
+// Copyright 2019 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under The General Public License (GPL), version 3.
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. Please review the Licences for the specific language governing
+// permissions and limitations relating to use of the SAFE Network Software.
+
+use super::Bootstrapped;
+use crate::{
+    error::{Result, RoutingError},
+    id::PublicId,
+    messages::{RoutingMessage, SignedMessage},
+    routing_table::Authority,
+    time::Instant,
+};
+use std::collections::BTreeSet;
+
+pub trait Unrelocated: Bootstrapped {
```

Comment:
It looks fine, but what is the benefit of using a trait over a free function.
Is there a problem with passing the PublicId?

Reply:
It's mostly about making the borrow checker happy. I'd prefer to use either free functions or composition but it would require more extensive refactoring. Maybe something for the future?